spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java

```diff
@@ -58,10 +60,21 @@ public ReactorServerHttpRequest(HttpServerRequest request, NettyDataBufferFactor
 	private static URI initUri(HttpServerRequest channel) {
 		Assert.notNull(channel, "'channel' must not be null");
 		InetSocketAddress address = channel.remoteAddress();
-		String requestUri = channel.uri();
+		String requestUri = squashSlashes(channel.uri());
 		return (address != null ? getBaseUrl(address).resolve(requestUri) : URI.create(requestUri));
 	}
 
+	/**
+	 * Remove redundant slashes from request URI-string
+	 *
+	 * <p>
+	 *   http://example.com/// -> http://example.com/
+	 * </p>
+	 */
+	private static String squashSlashes(String uri) {
+		return isNull(uri) ? uri : uri.replaceAll("/{2,}", "/");
```

Comment:
you are talking about URL, but npt URI, with actually i'm replacing, so answering on your question: no, it wouldn't collapse `http://x.y.z///` to `http:/x.y.z/`, instead, it should collapse into `http://x.y.z/`